rocks/RocksDatabase.java

```diff
@@ -189,27 +190,22 @@ private void initialiseData() {
         } catch (RocksDBException e) {
             throw TypeDBException.of(e);
         }
+        mayInitRocksLogger();
     }
 
     protected void load() {
-        loadSchema();
+        initialiseOrLoadSchema();
+        validateEncodingVersion();
         loadData();
-        mayInitRocksLogger();
         isOpen.set(true);
         try (RocksSession.Schema session = createAndOpenSession(SCHEMA, new Options.Session()).asSchema()) {
             try (RocksTransaction.Schema txn = session.initialisationTransaction()) {
-                validateEncoding(txn.schemaStorage());
```

Comment:
similar to initialisation, we do the validation right after loading the schema database, and then proceed